frontend/js/marc-record-merge-postmerge-service.js

```diff
@@ -256,9 +256,15 @@ export function mergeUniqueF042(preferredRecord, otherRecord, mergedRecordParam)
   const mergedRecord = new MarcRecord(mergedRecordParam);
   const f042s = mergedRecord.get(/^042$/);
   if (f042s.length > 0) {
-    const uniqueValues = [...new Set([].concat(...f042s.map(field => field.subfields.map(sub => sub.value))))];
-    const uniqueSubs = uniqueValues.map(value => {return {code: 'a', value}})
+    // Collect only values of subfields with code a
+    const arrayContainingArrayofSubValues = f042s.map(field => field.subfields.filter(sub => sub.code === 'a').map(sub => sub.value));
+    // Flatten array and make Set to filter only unique values
+    const setOfUniqueSubValues = new Set([].concat(...arrayContainingArrayofSubValues));
+    // Transform Set to array and reconstruct subfields
+    const uniqueSubs = [...setOfUniqueSubValues].map(value => {return {code: 'a', value}})
+    // Remove existing 042 fields from merged record
```

Comment:
042-fields contain only a-subfields according to official MARC spec but other custom subfields might be present. Better to make this foolproof:
```js
const subfields = someFunction();

function someFunction() {
  const subfields = f042s.map(({subfields}) => subfields).flat();

  return subfields.reduce((acc, ({code, value})) => {
    return acc.some(exists) ? acc : acc.concat({code, value});

    function exists(({code: existingCode, value: existingValue}) {
      return existingCode === code && existingValue === value;
    }
  }, []);
}
```